vlayer/ethereum/circuits/lib/src/README.md

```diff
@@ -0,0 +1,49 @@
+# Slots
+
+This is a [Noir](https://noir-lang.org) library for slots.
+
+The library consists of functions:
+
+- ```rust
+  pub(crate) fn mapping(slot: Bytes32, key: Bytes32) -> Bytes32;
+  ```
+
+  Calculates slot number of value corresponding to a mapping `key`. `Slot` argument is the storage location of the mapping calculated after applying storage layout rules.
+
+- ```rust
+  pub(crate) fn dynamic_array(slot: Bytes32, size: Field, index: Field) -> Bytes32;
+  ```
+
+  Calculates slot number where element at `index` in dynamic array is located. `Slot` argument is the storage location of the dynamic array calculated after applying storage layout rules. `Size` defines the size of elements in the array.
+
+- There is an analogical function where `slot` argument is a precalculated slot of the first element in the array:
```

Comment:
Let's describe - when is the other function useful. Some example code in Sol.

Reply:
Actually there is an example of usage for this function below.